normalize-own-data/quantile_normalize_own_data.Rmd

```diff
@@ -0,0 +1,207 @@
+We can read this data into R with the URL directly later on in this notebook.
+
+```{r}
+qn_url <- qn_target_info$s3_url
+```
+
+This is all the information we need to continue.
+
+```{r}
+rm(qn_target_info)
+```
+
+## Set up 
+
+We need the [`preprocessCore`](https://bioconductor.org/packages/release/bioc/html/preprocessCore.html) 
+package to perform quantile normalization. 
+
+```{r}
+if (!("preprocessCore" %in% installed.packages())) {
+  BiocManager::install("preprocessCore", update = FALSE)
+}
+```
+
+```{r}
+`%>%` <- dplyr::`%>%`
+```
+
+### Read in data
+
+First, the gene expression matrix.
+
+```{r}
+expression_file <- file.path("data", "GSE39842_RMA_converted.tsv")
+expression_df <- readr::read_tsv(expression_file)
+```
+```{r}
+expression_matrix <- expression_df %>%
+  # filter out any of the control probes
+  dplyr::filter(!grepl("AFFX", ENSEMBL)) %>%
+  as.data.frame() %>%
+  tibble::column_to_rownames("ENSEMBL") %>%
+  as.matrix()
+```
+
+Now, the QN target -- this will ultimately need to be a vector.
+
+```{r}
+qn_target <- as.vector(as.matrix(readr::read_tsv(qn_url,
+                                                 col_names = FALSE)))
+```
+
+## Quantile normalization
+
+```{r}
+qn_matrix <- preprocessCore::normalize.quantiles.use.target(expression_matrix,
+                                                            qn_target)
+```
+
+To write this to file or to do plotting with `ggplot2`, we'll want to neaten 
+things a bit (e.g., use column names).
+
+```{r}
+colnames(qn_matrix) <- colnames(expression_df)[-1]
+qn_df <- data.frame(Gene = rownames(expression_matrix),
+                    qn_matrix)
+```
+
+### Plotting
+
+Let's make some density plots to demonstrate the effect of quantile 
+normalization.
+
+#### Before
+
+```{r}
+# we need this information in long format to make the plots
+exprs_long_df <- expression_df %>% 
+  reshape2::melt() %>%
+  dplyr::mutate(Sample = variable) %>%
+  dplyr::select(-variable)
+```
+
+```{r}
+exprs_long_df %>%
+  ggplot2::ggplot(ggplot2::aes(x = value, fill = Sample)) +
+  ggplot2::geom_density(alpha = 0.2) +
+  ggplot2::theme_bw() +
+  ggplot2::labs(title = "Before quantile normalization",
+                x = "expression value")
+```
+
+The distributions look pretty similar, but they're not the _same_.
+
+#### After
+
+```{r}
+qn_long_df <- qn_df %>% 
+  reshape2::melt() %>%
+  # because the sample names were the column names, they are not pretty!
+  dplyr::mutate(Sample = gsub("X7", "7", 
+                              gsub("[[:punct:]]", " ", variable))) %>%
+  dplyr::select(-variable) 
+```
+
+```{r}
+qn_long_df %>%
+  ggplot2::ggplot(ggplot2::aes(x = value, fill = Sample)) +
+  ggplot2::geom_density(alpha = 0.2) +
+  ggplot2::theme_bw() +
+  ggplot2::labs(title = "After quantile normalization",
+                x = "expression value")
+```
+
+This looks like all the samples follow the same distribution.
+We can use `facet_wrap` to get a better look.
+
+```{r}
+qn_long_df %>%
```

Comment:
Could assign the plot to the name and then add the extra lines to it so you don't have to repeat yourself and further illustrate you are building on to the original plot.